tools/tpm2_createpolicy.c

```diff
@@ -0,0 +1,351 @@
+        LOG_ERR("Failed Policy Get Digest\n");
+        return rval;
+    }
+
+    LOG_INFO("Created Policy Digest:\n");
+
+    //save the policy buffer in a file for use later
+    bool result = files_save_bytes_to_file(ctx->policyfile, (UINT8 *) &ctx->policy_digest.t.buffer,
+            ctx->policy_digest.t.size);
+    if (!result) {
+        LOG_ERR("Failed to save policy digest into file \"%s\"",
+                ctx->policyfile);
+        return -1;
+    }
+
+    // Need to flush the session here.
+    rval = Tss2_Sys_FlushContext(ctx->sapi_context,
+            ctx->policy_session->sessionHandle);
+    if (rval != TPM_RC_SUCCESS) {
+        LOG_ERR("Failed Flush Context\n");
+        return rval;
+    }
+
+    // And remove the session from sessions table.
+    return tpm_session_auth_end(ctx->policy_session);
+}
+
+bool parse_policy_type(createpolicypcr_ctx *ctx, char *argv[]) {
+
+    if (ctx->pcr_flags.policy_type_pcr_flag) {
+        if (!ctx->policy_file_flag || !ctx->pcr_flags.pcr_index_flag) {
+            showArgMismatch(argv[0]);
+            return false;
+        } else {
+            LOG_INFO("Policy File = %s\n", ctx->policyfile);
+            LOG_INFO("PCR Index= %d\n", ctx->pcr_index);
+            UINT32 rval = build_policy(ctx, build_pcr_policy);
+            if (rval != TPM_RC_SUCCESS) {
+                LOG_ERR("Failed build_policy\n");
+                return false;
+            }
+        }
+    }
+    return true;
+}
+
+static bool init(int argc, char *argv[], createpolicypcr_ctx *ctx) {
+
+    struct option sOpts[] = { 
+        { "policy-file",    required_argument,  NULL,   'f' }, 
+        { "policy-pcr",     no_argument,        NULL,  'P' }, 
+        { "pcr-index",      required_argument,  NULL,   'i' }, 
+        { "pcr-alg",        required_argument,  NULL,   'g' }, 
+        { "pcr-input-file", required_argument,  NULL,   'F' }, 
+        { NULL,             no_argument,        NULL,   '\0'}, 
+    };
+
+    if (argc == 1) {
+        showArgMismatch(argv[0]);
+        return false;
+    }
+
+    if (argc > (int) (2 * sizeof(sOpts) / sizeof(struct option))) {
+        showArgMismatch(argv[0]);
+        return false;
+    }
+
+    int opt;
+    bool result;
+    while ((opt = getopt_long(argc, argv, "Pf:i:g:F:", sOpts, NULL)) != -1) {
+        switch (opt) {
+        case 'f':
+            ctx->policy_file_flag = true;
+            snprintf(ctx->policyfile, sizeof(ctx->policyfile), "%s", optarg);
+            break;
+        case 'P':
+            ctx->pcr_flags.policy_type_pcr_flag = true;
+            LOG_INFO("Policy type chosen is policyPCR.\n");
+            break;
+        case 'i':
+            ctx->pcr_flags.pcr_index_flag = true;
+            if (string_bytes_get_uint32(optarg, &ctx->pcr_index) != true) {
+                return false;
+            }
+            // pcr_index is unsigned... never can be less than 0
+            int max_supported_pcrs = get_max_supported_pcrs(ctx->sapi_context);
```

Comment:
I would make this a field in the context so you don't have to call this twice since TPM commands are slow, also I would check this outside of the getopt loop, so after getopt you can just always read the max, store it in context, and then if -i is specified check your bounds.